packages/flutter/lib/src/material/theme_data.dart

```diff
@@ -614,245 +646,54 @@ class ThemeData {
-
-  @override
-  Color get textSelectionHandleColor => delegate.textSelectionHandleColor;
-
-  @override
-  Color get unselectedWidgetColor => delegate.unselectedWidgetColor;
-
-  @override
-  TextTheme get primaryTextTheme => _primaryTextTheme ??= delegate.primaryTextTheme.merge(localTextGeometry);
-  TextTheme _primaryTextTheme;
-
-  @override
-  TextTheme get accentTextTheme => _accentTextTheme ??= delegate.accentTextTheme.merge(localTextGeometry);
-  TextTheme _accentTextTheme;
-
-  @override
-  TextTheme get textTheme => _textTheme ??= delegate.textTheme.merge(localTextGeometry);
-  TextTheme _textTheme;
-
-  /// This should be identical to [ThemeData.copyWith].
-  @override
-  ThemeData copyWith({
-    Brightness brightness,
-    Color primaryColor,
-    Brightness primaryColorBrightness,
-    Color accentColor,
-    Brightness accentColorBrightness,
-    Color canvasColor,
-    Color scaffoldBackgroundColor,
-    Color cardColor,
-    Color dividerColor,
-    Color highlightColor,
-    Color splashColor,
-    Color selectedRowColor,
-    Color unselectedWidgetColor,
-    Color disabledColor,
-    Color buttonColor,
-    Color secondaryHeaderColor,
-    Color textSelectionColor,
-    Color textSelectionHandleColor,
-    Color backgroundColor,
-    Color dialogBackgroundColor,
-    Color indicatorColor,
-    Color hintColor,
-    Color errorColor,
-    TextTheme textTheme,
-    TextTheme primaryTextTheme,
-    TextTheme accentTextTheme,
-    IconThemeData iconTheme,
-    IconThemeData primaryIconTheme,
-    IconThemeData accentIconTheme,
-    TargetPlatform platform,
-  }) {
-    return _copyThemeDataWith(
-      this,
-      brightness: brightness,
-      primaryColor: primaryColor,
-      primaryColorBrightness: primaryColorBrightness,
-      accentColor: accentColor,
-      accentColorBrightness: accentColorBrightness,
-      canvasColor: canvasColor,
-      scaffoldBackgroundColor: scaffoldBackgroundColor,
-      cardColor: cardColor,
-      dividerColor: dividerColor,
-      highlightColor: highlightColor,
-      splashColor: splashColor,
-      selectedRowColor: selectedRowColor,
-      unselectedWidgetColor: unselectedWidgetColor,
-      disabledColor: disabledColor,
-      buttonColor: buttonColor,
-      secondaryHeaderColor: secondaryHeaderColor,
-      textSelectionColor: textSelectionColor,
-      textSelectionHandleColor: textSelectionHandleColor,
-      backgroundColor: backgroundColor,
-      dialogBackgroundColor: dialogBackgroundColor,
-      indicatorColor: indicatorColor,
-      hintColor: hintColor,
-      errorColor: errorColor,
-      textTheme: textTheme,
-      primaryTextTheme: primaryTextTheme,
-      accentTextTheme: accentTextTheme,
-      iconTheme: iconTheme,
-      primaryIconTheme: primaryIconTheme,
-      accentIconTheme: accentIconTheme,
-      platform: platform,
-    );
+  bool operator ==(Object other) {
+    final _IdentityThemeDataCacheKey otherKey = other;
```

Comment:
and a comment that we are explicitly ignoring the possibility that the types might not match in the interests of speed

Reply:
Done.